ee/src/main/java/org/jboss/as/ee/subsystem/ContextServiceAdd.java

```diff
@@ -48,15 +45,18 @@ private ContextServiceAdd() {
 
     @Override
     protected void performRuntime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
-        ModelNode model = resource.getModel();
-        final String name = PathAddress.pathAddress(operation.get(ModelDescriptionConstants.ADDRESS)).getLastElement().getValue();
+        final ModelNode model = resource.getModel();
+        final String name = context.getCurrentAddressValue();
 
         final String jndiName = ContextServiceResourceDefinition.JNDI_NAME_AD.resolveModelAttribute(context, model).asString();
         final boolean useTransactionSetupProvider = ContextServiceResourceDefinition.USE_TRANSACTION_SETUP_PROVIDER_AD.resolveModelAttribute(context, model).asBoolean();
 
         // install the service which manages the default context service
         final ContextServiceService contextServiceService = new ContextServiceService(name, jndiName, new DefaultContextSetupProviderImpl());
-        final ServiceBuilder<ContextServiceImpl> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getContextServiceServiceName(name), contextServiceService);
+        final ServiceBuilder serviceBuilder = context.getCapabilityServiceTarget()
+                .addService(ContextServiceResourceDefinition.CAPABILITY.getCapabilityServiceName(name))
```

Comment:
Unless you want to refactor ContextServiceService to support installation as a MSC 1.4.x service, you will need to use:
``` java
ServiceBuilder<ContextServiceImpl> serviceBuilder = context.getCapabilityServiceTarget().addCapability(ContextServiceResourceDefinition.CAPABILITY, contextServiceService);
```

Reply:
done